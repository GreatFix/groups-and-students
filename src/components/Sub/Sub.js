import React from 'react'
import classes from './Sub.module.css'
import PropTypes from 'prop-types'

const Sub = (props) => {
  return <p className={`${classes.Sub} ${props.className}`}>{props.children}</p>
}

Sub.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
}

Sub.defaultProps = {
  className: '',
  children: null,
}

export default React.memo(Sub)
