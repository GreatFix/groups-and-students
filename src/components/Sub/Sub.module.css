.Sub {
  display: grid;
  justify-content: center;
  align-content: center;
  background-color: #54d8a8;
  border-radius: 2px;
  font-size: 1em;
  font-family: 'Roboto Slab', serif;
  padding: 5px;
  margin: 4px 2px 0 2px;
}
