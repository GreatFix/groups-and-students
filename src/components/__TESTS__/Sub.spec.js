import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Sub from '../Sub/Sub'

const TEST_TEXT = 'May the force be with you'
const TEST_CLASS = 'TEST_CLASS'

describe('Sub', () => {
  test('renders a text', () => {
    const { getByText } = render(<Sub>{TEST_TEXT}</Sub>)
    expect(getByText(TEST_TEXT)).toBeInTheDocument()
  })

  test('gets a class', () => {
    const { container } = render(<Sub className={TEST_CLASS}>{TEST_TEXT}</Sub>)
    expect(container.querySelector('.Sub')).toHaveClass(TEST_CLASS)
  })

  test('render properly', () => {
    const { container } = render(<Sub>{TEST_TEXT}</Sub>)
    expect(container).toMatchSnapshot()
  })
})
