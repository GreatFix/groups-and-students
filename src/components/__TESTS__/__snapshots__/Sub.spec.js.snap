// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sub render properly 1`] = `
<div>
  <p
    class="Sub "
  >
    May the force be with you
  </p>
</div>
`;
